gen/utils/data/data.go

```diff
@@ -0,0 +1,181 @@
+				err := ds.retrieveMetaFileFromRemote(className)
+				if err != nil {
+					return err
+				}
+			}
+		}
+	}
+
+	return nil
+}
+
+func (ds *DataStore) refreshMetaFiles() error {
+	// Check if the meta data should be refreshed from remote location as specified in the environment variable.
+	metaRefresh := os.Getenv(constEnvMetaRefresh)
+	// If the environment variable is not set, the default is to not refresh the meta data.
+	if metaRefresh != "" {
+		var refresh bool
+		var err error
+		// If the environment variable is set, parse it to a boolean value.
+		// If the parsing fails, log a warning and skip retrieval of meta data.
+		// Parsing accepts 1, t, T, TRUE, true, True, 0, f, F, FALSE, false, False.
+		// Any other value returns an error, which is logged and the retrieval is skipped.
+		refresh, err = strconv.ParseBool(metaRefresh)
+		if err != nil {
+			genLogger.Warn(fmt.Sprintf("Refreshing of meta is skipped due to error: %s.", err.Error()))
+		} else if refresh {
+			genLogger.Debug(fmt.Sprintf("Refreshing meta data from remote location: %s.", ds.metaHost))
+			for _, className := range utils.GetFileNamesFromDirectory(constMetaPath, true) {
+				// ENHANCEMENT: Concurrently retrieve/write the meta data.
+				// Retrieve the meta file for the class from the remote location.
+				// Only retrieve the meta file if it is not already retrieved.
+				if !slices.Contains(ds.retrievedClasses, className) {
+					err = ds.retrieveMetaFileFromRemote(className)
+					if err != nil {
+						return err
+					}
+				}
+			}
+			genLogger.Debug(fmt.Sprintf("Successfully refreshed meta data from remote location: %s.", ds.metaHost))
+		}
+	}
+	return nil
+}
+
+func (ds *DataStore) retrieveMetaFileFromRemote(className string) error {
+	packageName, shortName, err := splitClassNameToPackageNameAndShortName(className)
+	if err != nil {
+		return err
+	}
+	url := fmt.Sprintf(constMetaFileUrl, ds.metaHost, packageName, shortName)
+	genLogger.Debug(fmt.Sprintf("Retrieving meta data for class '%s' from: %s.", className, url))
+
+	res, err := ds.client.Get(url)
+	if err != nil {
+		genLogger.Error(fmt.Sprintf("Error during retrieval of meta file for class '%s': %s.", className, err.Error()))
+		return err
+	}
+
+	var buffer bytes.Buffer
+	_, err = io.Copy(&buffer, res.Body)
+	if err != nil {
+		genLogger.Error(fmt.Sprintf("Error during reading of file for class '%s': %s.", className, err.Error()))
+		return err
+	}
+
+	outputFile, err := os.Create(fmt.Sprintf("%s/%s.json", constMetaPath, className))
+	if err != nil {
+		genLogger.Error(fmt.Sprintf("Error during creation of file for class '%s': %s.", className, err.Error()))
+		return err
+	}
+
+	defer outputFile.Close()
+	_, err = outputFile.Write(buffer.Bytes())
+	if err != nil {
+		genLogger.Error(fmt.Sprintf("Error during writing to file for class '%s': %s.", className, err.Error()))
+		return err
+	}
+
+	ds.retrievedClasses = append(ds.retrievedClasses, className)
```

Comment:
Any risk of duplication in this list?

Reply:
in theory it could, so moved the conditional inside the retrieveMetaFileFromRemote function